mlflow/store/azure_blob_artifact_repo.py

```diff
@@ -0,0 +1,114 @@
+import os
+import re
+
+from six.moves import urllib
+
+from mlflow.entities.file_info import FileInfo
+from mlflow.store.artifact_repo import ArtifactRepository
+from mlflow.utils.file_utils import build_path, get_relative_path, TempDir
+
+
+class AzureBlobArtifactRepository(ArtifactRepository):
+    """
+    Stores artifacts on Azure Blob Storage.
+
+    This repository is used with URIs of the form
+    ``wasbs://<container-name>@<ystorage-account-name>.blob.core.windows.net/<path>``,
+    following the same URI scheme as Hadoop on Azure blob storage. It requires that your Azure
+    storage access key be available in the environment variable ``AZURE_STORAGE_ACCESS_KEY``.
+    """
+
+    def __init__(self, artifact_uri, client=None):
+        if client:
+            self.client = client
+        else:
+            from azure.storage.blob import BlockBlobService
+            (_, account, _) = AzureBlobArtifactRepository.parse_wasbs_uri(artifact_uri)
+            self.client = BlockBlobService(account_name=account,
+                                           account_key=os.environ.get("AZURE_STORAGE_ACCESS_KEY"))
+        super(AzureBlobArtifactRepository, self).__init__(artifact_uri)
+
+    @staticmethod
+    def parse_wasbs_uri(uri):
```

Comment:
Could we at least unit test this method? That way we have at least have a basis to add more unit tests for the rest of the BlockBlobService (which I suspect is actually not _that_ bad to mock out with `mock.MagicMock`)

Reply:
Done